src/io.cpp

```diff
@@ -519,88 +461,55 @@ Pipeline::Pipeline(CentroidAlgorithm *centroidAlgorithm,
-            centroidChoice.Register("cog", "Center of Gravity Centroid Algorithm",
-                                    CoGCentroidAlgorithmPrompt);
-            centroidChoice.Register("iwcog", "Iterative Weighted Center of Gravity Algorithm",
-                                    IWCoGCentroidAlgorithmPrompt);
-
-            result.centroidAlgorithm = std::unique_ptr<CentroidAlgorithm>(
-                (centroidChoice.Prompt("Choose centroid algo"))());
-            break;
-        }
-
-        case PipelineStage::CentroidMagnitudeFilter: {
-            result.centroidMinMagnitude = Prompt<int>("Minimum magnitude");
-            break;
-        }
-
-        case PipelineStage::Database: {
-            std::string path = Prompt<std::string>("Database file");
-            std::fstream fs;
-            fs.open(path, std::fstream::in | std::fstream::binary);
-            fs.seekg(0, fs.end);
-            long length = fs.tellg();
-            fs.seekg(0, fs.beg);
-            std::cerr << "Reading " << length << " bytes of database" << std::endl;
-            result.database = std::unique_ptr<unsigned char[]>(new unsigned char[length]);
-            fs.read((char *)result.database.get(), length);
-            std::cerr << "Done" << std::endl;
-            break;
-        }
-
-        case PipelineStage::StarId: {
-            InteractiveChoice<StarIdAlgorithmFactory> starIdChoice;
-            starIdChoice.Register("dummy", "Random", DummyStarIdAlgorithmPrompt);
-            starIdChoice.Register("gv", "Geometric Voting", GeometricVotingStarIdAlgorithmPrompt);
-            starIdChoice.Register("pyramid", "Pyramid Scheme", PyramidStarIdAlgorithmPrompt);
-
-            result.starIdAlgorithm = std::unique_ptr<StarIdAlgorithm>(
-                (starIdChoice.Prompt("Choose Star-ID algo"))());
-            break;
-        }
-
-        case PipelineStage::AttitudeEstimation: {
-            InteractiveChoice<AttitudeEstimationAlgorithmFactory> attitudeEstimationAlgorithmChoice;
-            attitudeEstimationAlgorithmChoice.Register("dqm", "Davenport Q Method", DavenportQAlgorithmPrompt);
-            result.attitudeEstimationAlgorithm = std::unique_ptr<AttitudeEstimationAlgorithm>(
-                (attitudeEstimationAlgorithmChoice.Prompt("Choose Attitude algo"))());
-            break;
-        }
 
-        case PipelineStage::Done: {
-            // fuck style guides
-            goto PipelineDone;
-        }
-        }
+    // centroid algorithm stage
+    if (values.centroidAlgo == "dummy") {
+        result.centroidAlgorithm = std::unique_ptr<CentroidAlgorithm>(new DummyCentroidAlgorithm(values.dummyCentroidNumStars));
+    } else if (values.centroidAlgo == "cog") {
+        result.centroidAlgorithm = std::unique_ptr<CentroidAlgorithm>(new CenterOfGravityAlgorithm());
+    } else if (values.centroidAlgo == "iwcog") {
+        result.centroidAlgorithm = std::unique_ptr<CentroidAlgorithm>(new IterativeWeightedCenterOfGravityAlgorithm());
+    }
```

Comment:
Add another else statement here to check if values.centroidAlgo is set to an illegal value, similarly for other algos